pkg/webservice/handlers_test.go

```diff
@@ -678,36 +681,126 @@ func TestGetNodesUtilJSON(t *testing.T) {
+	var errInfo dao.YAPIError
+	err = json.Unmarshal(resp.outputBytes, &errInfo)
+	assert.NilError(t, err, "getNodeUtilisation should have returned and error")
+
+	partition := setup(t, configDefault, 1)
+	utilisation := &dao.NodesUtilDAOInfo{}
+	err = json.Unmarshal(resp.outputBytes, utilisation)
+	assert.NilError(t, err, "getNodeUtilisation should have returned an empty object")
+	assert.Equal(t, utilisation.ResourceType, "", "unexpected type returned")
+	assert.Equal(t, len(utilisation.NodesUtil), 0, "no nodes should be returned")
+	assert.Assert(t, confirmNodeCount(utilisation.NodesUtil, 0), "unexpected number of nodes returned should be 0")
+
+	// create test nodes
+	nodeRes := resources.NewResourceFromMap(map[string]resources.Quantity{"first": 10}).ToProto()
+	nodeRes2 := resources.NewResourceFromMap(map[string]resources.Quantity{"first": 10, "second": 5}).ToProto()
+	node1ID := "node-1"
+	node1 := objects.NewNode(&si.NodeInfo{NodeID: node1ID, SchedulableResource: nodeRes})
+	node2ID := "node-2"
+	node2 := objects.NewNode(&si.NodeInfo{NodeID: node2ID, SchedulableResource: nodeRes2})
+
+	err = partition.AddNode(node1, nil)
+	assert.NilError(t, err, "add node to partition should not have failed")
+	err = partition.AddNode(node2, nil)
+	assert.NilError(t, err, "add node to partition should not have failed")
 
 	// get nodes utilization
-	res1 := getNodesUtilJSON(partition, siCommon.Memory)
-	res2 := getNodesUtilJSON(partition, siCommon.CPU)
-	res3 := getNodesUtilJSON(partition, "GPU")
-	resNon := getNodesUtilJSON(partition, "non-exist")
-	subres1 := res1.NodesUtil
-	subres2 := res2.NodesUtil
-	subres3 := res3.NodesUtil
-	subresNon := resNon.NodesUtil
-
-	assert.Equal(t, res1.ResourceType, siCommon.Memory)
-	assert.Equal(t, subres1[2].NumOfNodes, int64(1))
-	assert.Equal(t, subres1[4].NumOfNodes, int64(1))
-	assert.Equal(t, subres1[2].NodeNames[0], node2ID)
-	assert.Equal(t, subres1[4].NodeNames[0], node1ID)
-
-	assert.Equal(t, res2.ResourceType, siCommon.CPU)
-	assert.Equal(t, subres2[2].NumOfNodes, int64(1))
-	assert.Equal(t, subres2[4].NumOfNodes, int64(1))
-	assert.Equal(t, subres2[2].NodeNames[0], node1ID)
-	assert.Equal(t, subres2[4].NodeNames[0], node2ID)
-
-	assert.Equal(t, res3.ResourceType, "GPU")
-	assert.Equal(t, subres3[4].NumOfNodes, int64(1))
-	assert.Equal(t, subres3[4].NodeNames[0], node2ID)
-
-	assert.Equal(t, resNon.ResourceType, "non-exist")
-	assert.Equal(t, subresNon[0].NumOfNodes, int64(0))
-	assert.Equal(t, len(subresNon[0].NodeNames), 0)
+	getNodeUtilisation(resp, req)
+	utilisation = &dao.NodesUtilDAOInfo{}
+	err = json.Unmarshal(resp.outputBytes, utilisation)
+	assert.NilError(t, err, "getNodeUtilisation should have returned an object")
+	assert.Equal(t, utilisation.ResourceType, "", "unexpected type returned")
+	assert.Equal(t, len(utilisation.NodesUtil), 10, "empty usage: unexpected bucket count returned")
+	assert.Assert(t, confirmNodeCount(utilisation.NodesUtil, 0), "unexpected number of nodes returned should be 2")
```

Comment:
nit:"unexpected number of nodes returned should be 0"

Reply:
fixed